CMakeLists.txt

```diff
@@ -40,17 +59,20 @@ file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/
 # ---- Create library ----
 
 # Note: for header-only libraries change all PUBLIC flags to INTERFACE and create an interface
-# target: add_library(Greeter INTERFACE) set_target_properties(Greeter PROPERTIES
-# INTERFACE_COMPILE_FEATURES cxx_std_17)
+# target: add_library(Greeter INTERFACE)
+add_library(Greeter)
+target_compile_features(Greeter PUBLIC cxx_std_17)
 
-add_library(Greeter ${headers} ${sources})
-
-set_target_properties(Greeter PROPERTIES CXX_STANDARD 17)
+target_sources(Greeter PRIVATE ${headers} ${sources})
```

Comment:
Tbh I don't see an advantage in splitting `add_library()` to `add_library` and `target_sources`, as sources are known at this point anyways. Let's try to keep the CMake as simple as possible.

Reply:
OK